src/Native/build.proj

```diff
@@ -89,6 +89,8 @@
                           RelativePath="Microsoft.ML\runtimes\$(PackageRid)\native" />
       <NativePackageAsset Include="$(NativeAssetsBuiltPath)\$(NativeLibPrefix)LdaNative$(NativeLibExtension)"
                           RelativePath="Microsoft.ML\runtimes\$(PackageRid)\native" />
+      <NativePackageAsset Include="$(NativeAssetsBuiltPath)\$(NativeLibPrefix)LIBMFNative$(NativeLibExtension)"
+                          RelativePath="Microsoft.ML\runtimes\$(PackageRid)\native" />
```

Comment:
Most users won't need `libmf` right? I don't think we should be packaging it into our main NuGet package: `Microsoft.ML`

Reply:
Do you have any example or document? It's not quite clear to me how to modify the build framework.

Reply:
Check out how the `HalLearners` project is built:

https://github.com/dotnet/machinelearning/search?p=1&q=HalLearners&unscoped_q=HalLearners

Reply:
To your original question, I am not sure. A [package](https://github.com/benfred/implicit/) with three algorithms for recommender systems can have more than 1k stars. My feeling is that shipping the recommender module with the main nuget might not be that bad.

Reply:
I disagree. We should be limiting the main nuget package to things that `> 80%` of our customers will need. We should be splitting the "optional" pieces off into separate nuget packages, so the customers who don't use those optional pieces don't need to install those pieces on their end user's machines. It bloats their apps for no reason.

Reply:
I see. That's ok to me. Please do take a look at my changes to see if I am doing what you expect. This is my first time of using nuget. We have recommender module in C# and LIBMF in C++. Now, I (hopefully) have made LIBMF a nuget and the C# recommender module consumes that nuget.